extensions/wikia/MercuryApi/MercuryApiController.class.php

```diff
@@ -467,7 +465,30 @@ public function getCuratedContentSection() {
 			throw new NotFoundApiException( 'No members' );
 		}
 
-		$this->response->setVal( 'items', $data[ 'items' ] );
+		$this->response->setVal( 'items', $data['items'] );
+	}
+
+	public function getMainPageDetailsAndAdsContext() {
+		$mainPageTitle = Title::newMainPage();
+		$mainPageArticleID = $mainPageTitle->getArticleID();
+		$article = Article::newFromID( $mainPageArticleID );
+
+		$articleDetails = $this->getArticleDetails( $article );
+
+		if ( empty( $articleDetails ) ) {
+			throw new WikiaException( 'Article details empty', 500 );
```

Comment:
You already check if `details` are set in https://github.com/Wikia/mercury/pull/1538/files#diff-88b7fe5a8a84ed29d43821b2f75c6f8fR54

I don't see the need to throw exception here as it will prevent us from delivering adsContext without good reason. IMO setting them to null is good enough.